### ISSUE TITLE
Can't "Save as Copy" items that have a relationships (o2m, m2o, m2m) as non-admin user

### ISSUE THREAD

**Issue by KnowHowHere** [directus/api#724]
# Bug Report

## Steps to Reproduce

1. Create a collection with a relationship (one-to-many or many-to-many)
2. Create a new item as a _non-admin_ user
3. Revisit item, make an edit and choose "Save as Copy"
4. See error:
  Updating item from "directus_users" collection was denied
5. Allow update Full permission to directus_users for role
6. See error:
  Updating item from "directus_user_roles" collection was denied
7. Allow update Full permission to directus_user_roles for role
8. See error:
  You are not allowed to create, update or delete roles
9. Comment out lines 591 - 593 of CoreServiceProvider.php
```
  // if (!$acl->isAdmin()) {
  //       throw new ForbiddenException('You are not allowed to create, update or delete roles');
  // }
```
10. Copy saved

## Expected Behaviour
Should be able to copy items without Directus trying to write to the user or user roles table

## Actual Behaviour
It appears that the API tries to rewrite/reset the full directus_users and directus_users_roles records for the current user with their current values. i.e no changes, just rewriting all fields with their current values.

## Technical Details

- Device: Desktop
- OS: OS X
- Web Server: NGINX
- PHP Version:7.3
- Database: MariaDB
- Install Method: cloned `master` branch [c6b1ed8]

**Comment by WellingGuzman** [comment]
Hey @KnowHowHere, You can, the problem here seems to be about permissions. If the user doesn't have permission to a related table the error will be thrown.

This might be related to the application sending the `role` data when copying a collection that have a user field.

> 9. Comment out lines 591 - 593 of CoreServiceProvider.php

It's hard-coded to allow only admin to change roles data on users. (Sending this data will be treated as changing)

ping @benhaynes @rijkvanzanten

**Comment by rijkvanzanten** [comment]
The app is not updating a user role when you make a copy of an item

**Comment by WellingGuzman** [comment]
@KnowHowHere is any of the related collection/field has relation with `directus_users`?

**Comment by KnowHowHere** [comment]
Hi there, no, no relations to directus_users and the role has full access to all tables, I'll setup a test item.create:before hook and supply so you can see the updates coming through for directus_users and directus_user_roles.

**Comment by WellingGuzman** [comment]
@KnowHowHere Can you share a database dump we can use to reproduce this issue?

**Comment by KnowHowHere** [comment]
Please find a simple example attached.  If you login with:

username: contributor@example.com
password: secret

and try to duplicate the location "Sydney Harbour Bridge" you should see the errors described.

The record _is_ duplicated, but because the app tries to then update the users and user_roles tables, which fails for non-admin users, it doesn't actually transition you to the new item ready for editing further as usual and as noted in #1276 the many-to-many table isn't duplicated, so if you go back out to the listing and open the new item directly, you see the notifications are empty.

If you want to login as an admin you can use:

username: admin@example.com
password: secret

Let me know if you need anything else.

[directus_example.sql.gz](https://github.com/directus/api/files/2799871/directus_example.sql.gz)

**Comment by WellingGuzman** [comment]
Thanks @KnowHowHere, I was able to duplicated the issue.

**Comment by WellingGuzman** [comment]
The issue here is that that the there's some field related to `directus_users`: `user_created` and `user_modified`.

The user with email `contributor@example.com` belongs to the `Contributor` role that doesn't have create permission on `directus_users`.

While the main item fields related with `directus_users` has not data, except for the `id`, the other related fields has related  fields related to `directus_fields` on their own, and those have more data than the `id`, meaning that user is not allowed to create new user data.

In short: The application is sending user data for a user that doesn't have permission to.

Any thoughts @rijkvanzanten?

**Comment by rijkvanzanten** [comment]
I'm not fully following what you're saying. What wrong JSON is the app sending to the API?

**Comment by rijkvanzanten** [comment]
So when duplicating an item we're removing the primary key, the user_created, user_modified, date_created, and date_modified values 👍
